### PR TITLE
feat(sdk-coin-stx): add support to nakamoto updgrade stack-stx tx

### PR DESCRIPTION
**Pull request by rliberal-simtlix**
EA-3369

TICKET: EA-3369

### CODE REVIEW

**Review by OttoAllmendinger — COMMENTED** [review]
flush - insufficient context

**Comment by rliberal-simtlix** [comment]
> flush - insufficient context
Breaking changes on STX for Nakamoto fork:

Fork:
https://docs.stacks.co/nakamoto-upgrade/nakamoto

Breaking changes in stack-stx staking adds new 4 parameters to contract call and changes from version 3 to 4 POX contract for staking.

https://docs.stacks.co/nakamoto-upgrade/signing-and-stacking/stacking-flow#solo-stacker-flow


TDD:
https://docs.google.com/document/d/1UBqC_uy2hQs8YW7y2Md6i03Ze2aWvAXVUgVj3S5SJrE/edit?pli=1#heading=h.pe165sdobg6x

**Review by theMobiusStrip — DISMISSED** [review]

File: modules/sdk-core/src/bitgo/staking/iStakingWallet.ts [hunk 1]
```diff
@@ -51,13 +51,30 @@ export interface StakeOptions {
    */
   blsSignature?: string;
   /**
-   * coin sepcific staking subtype
+   * stx coin specific staking subtype
```
Comment (evanzbitgo) [inline modules/sdk-core/src/bitgo/staking/iStakingWallet.ts:54]: this is not only for stx

File: modules/sdk-core/src/bitgo/staking/iStakingWallet.ts [hunk 2]
```diff
@@ -51,13 +51,30 @@ export interface StakeOptions {
...
+  // https://bitgoinc.atlassian.net/browse/EA-3482
+  /**
+   * pox-contract name (valid values are pox-3 and pox-4)
+   */
+  contractName?: 'pox-3' | 'pox-4';
+
+  /**
+   * stx signer signature
+   */
+  signerSignature?: string;
```
Comment (evanzbitgo) [inline modules/sdk-core/src/bitgo/staking/iStakingWallet.ts:77]: could we also update interface docs and also put `signerPub` and `signerSignature` next to each other?

**Review by theMobiusStrip — DISMISSED** [review]

**Review by theMobiusStrip — APPROVED** [review]

**Review by alebusse — APPROVED** [review]

**Pull request merged**